src/plugins/template/backend/int_executable.cpp

```diff
@@ -41,6 +41,44 @@ class TemporaryOverrideOutputs {
     }
 };
 
+inline void update_output_tensors(const ov::TensorVector& output_values, const HostTensorVector& outputs) {
+    OPENVINO_ASSERT(output_values.size() == outputs.size());
+    for (int i = 0; i < output_values.size(); ++i) {
+        auto& tensor = output_values[i];
+        auto& hosttensor = outputs[i];
+        if (hosttensor->get_is_allocated()) {
+            hosttensor->set_shape(tensor.get_shape());
+            std::copy_n(static_cast<uint8_t*>(tensor.data()),
+                        tensor.get_byte_size(),
+                        hosttensor->get_data_ptr<uint8_t>());
+        } else {
+            hosttensor->initialize(
+                make_shared<ov::op::v0::Constant>(tensor.get_element_type(), tensor.get_shape(), tensor.data()));
+        }
+    }
+}
+
+inline ov::TensorVector convert_hosttensors_2_tensors(const HostTensorVector& host_tensors, bool copy_data) {
+    ov::TensorVector ret_value;
+    ov::Tensor tensor;
+    for (const auto& hosttensor : host_tensors) {
+        if (hosttensor->get_element_type().is_dynamic()) {
+            tensor = ov::Tensor();
+        } else if (hosttensor->get_partial_shape().is_dynamic()) {
+            tensor = ov::Tensor(hosttensor->get_element_type(), {0});
```

Comment:
```suggestion
            tensor = ov::Tensor(hosttensor->get_element_type(), {max_int, 0});
```
Maybe if we will use empty tensor with max_int value it will allow us to differentiate dynamic tensor and empty {0} tensor

Reply:
Changed 😊